haystack/document_stores/pinecone.py

```diff
@@ -459,69 +690,163 @@ def get_documents_by_id(
+                    embedding_matrix.append(result["vectors"][_id]["values"])
+            if return_embedding:
+                values = embedding_matrix
+            else:
+                values = None
+            document_batch = self._get_documents_by_meta(
+                vector_id_matrix, meta_matrix, values=values, index=index, return_embedding=return_embedding
+            )
+            documents.extend(document_batch)
 
         return documents
 
+    def get_document_by_id(
+        self,
+        id: str,
+        index: Optional[str] = None,
+        headers: Optional[Dict[str, str]] = None,
+        return_embedding: Optional[bool] = None,
+        namespace: str = None,
+    ) -> Document:
+        """
+        Returns a single Document retrieved using an ID.
+
+        :param id: ID string to retrieve.
+        :param index: Optional index name for where to retrieve all documents from.
+        :param headers: Pinecone does not support headers.
+        :param return_embedding: Optional flag to return the embedding of the document.
+        :param namespace: Optional namespace to retrieve documents from.
+        """
+        documents = self.get_documents_by_id(
+            ids=[id], namespace=namespace, index=index, headers=headers, return_embedding=return_embedding
+        )
+        return documents[0]
+
     def get_embedding_count(
         self, index: Optional[str] = None, filters: Optional[Dict[str, Union[Dict, List, str, int, float, bool]]] = None
     ) -> int:
         """
         Return the count of embeddings in the document store.
+
+        :param index: Optional index name for where to retrieve all documents from.
+        :param filters: Filters are not supported for `get_embedding_count` in Pinecone.
         """
         if filters:
             raise NotImplementedError("Filters are not supported for get_embedding_count in PineconeDocumentStore")
 
-        index = index or self.index
-        index = self._sanitize_index_name(index)
-        if not self.pinecone_indexes.get(index, False):
-            raise ValueError(f"No index named {index} found in Pinecone.")
+        index = str(index or self.index)
+        index = _sanitize_index_name(index)
+        if index not in self.pinecone_indexes:
+            raise DocumentStoreError(
+                f"Index named '{index}' does not exist. Try reinitializing PineconeDocumentStore() and running "
+                f"'update_embeddings()' to create and populate an index."
+            )
 
         stats = self.pinecone_indexes[index].describe_index_stats()
-        # if no namespace return zero
-        count = stats["namespaces"][""]["vector_count"] if "" in stats["namespaces"] else 0
+        # if no embeddings namespace return zero
+        count = (
+            stats["namespaces"][self.embedding_namespace]["vector_count"]
+            if self.embedding_namespace in stats["namespaces"]
+            else 0
+        )
```

Comment:
:scroll: This looks so much like a tuple comprehension that I thought it was a bug :smile: Let's split it.
```suggestion
        if self.embedding_namespace in stats["namespaces"]:
            count = stats["namespaces"][self.embedding_namespace]["vector_count"]
        else:
            count = 0
```